src/js/msp/MSPConnector.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+var MSPConnectorImpl = function () {
+    this.baud = undefined;
+    this.port = undefined;
+    this.onConnectCallback = undefined;
+    this.onTimeoutCallback = undefined;
+    this.onDisconnectCallback = undefined;
+};
+
+MSPConnectorImpl.prototype.connect = function (port, baud, onConnectCallback, onTimeoutCallback, onFailureCallback) {
+
+    var self = this;
+    self.port = port;
+    self.baud = baud;
+    self.onConnectCallback = onConnectCallback;
+    self.onTimeoutCallback = onTimeoutCallback;
+    self.onFailureCallback = onFailureCallback;
+
+    serial.connect(self.port, {bitrate: self.baud}, function (openInfo) {
+        if (openInfo) {
+            var disconnectAndCleanup = function() {
+                serial.disconnect(function(result) {
+                    console.log('Disconnected');
+                    
+                    MSP.clearListeners();
+                    
+                    self.onTimeoutCallback();
+                });
+                
+                MSP.disconnect_cleanup();
+            };
+            
+            FC.resetState();
+            
+            // disconnect after 10 seconds with error if we don't get IDENT data
+            GUI.timeout_add('msp_connector', function () {
+                if (!CONFIGURATOR.connectionValid) {
+                    GUI.log(i18n.getMessage('noConfigurationReceived'));
+                    
+                    disconnectAndCleanup();
+                }
+            }, 10000);
+
+            serial.onReceive.addListener(read_serial);
+            
+            MSP.listen(update_packet_error);
+            mspHelper = new MspHelper();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['mspHelper' is not defined.](https://app.codacy.com/gh/emuflight/EmuConfigurator/pullRequest?prid=5626377)